packages/vector_graphics_compiler/lib/src/svg/parser.dart

```diff
@@ -1,30 +1,59 @@
 import 'dart:collection';
 
-import 'package:vector_graphics_compiler/src/vector_instructions.dart';
 import 'package:xml/xml_events.dart';
 
 import '../geometry/basic_types.dart';
 import '../geometry/matrix.dart';
 import '../geometry/path.dart';
-import 'node.dart';
 import '../paint.dart';
-
+import '../vector_instructions.dart';
 import 'colors.dart';
+import 'node.dart';
 import 'numbers.dart';
 import 'parsers.dart';
 import 'theme.dart';
 import 'xml.dart';
 
 /// An interface for building up a stack of vector commands.
-abstract class DrawCommandBuilder {
+class DrawCommandBuilder {
+  final List<Paint> _paints = <Paint>[];
+  final List<Path> _paths = <Path>[];
+  final List<DrawCommand> _commands = <DrawCommand>[];
+
   /// Add a save layer to the command stack.
-  void addSaveLayer(Paint paint) {}
+  void addSaveLayer(Paint paint) {
+    _commands
+        .add(DrawCommand(-1, _paints.length, DrawCommandType.saveLayer, null));
```

Comment:
nit: maybe rather than a -1 sentinel all over, just have these be named parameters and the Ids can be nullable when they're not used (or -1 in the ctor).

Reply:
If I do this I need to re-write a few hundred constructors, I'd rather not

Reply:
I have regex powers.

Reply:
Regex ...  It's not a story the Jedi would tell you